oak/server/time/roughtime_util_test.cc

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright 2020 The Project Oak Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "oak/server/time/roughtime_util.h"
+
+#include "gtest/gtest.h"
+
+namespace oak {
+
+TEST(RoughtimeUtil, TestValidOverlapOneOfThree) {
+  std::vector<RoughtimeInterval> intervals{{1, 4}, {2, 6}, {3, 5}};
```

Comment:
Could we add a couple of variants of this test with the input intervals re-ordered?

Reply:
Added the reversed versions of the first 3 tests.